### PR TITLE
MM-30436 Initialize server configuration with custom defaults

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-30436

Updated:
- Admin Guide > Configure Mattermost > Configure Settings > added new subsection: Load Custom Configuration Defaults immediately following the Environment Variables section

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -26,13 +26,21 @@ The name of the environment variable for any setting can be derived from the nam
...
 
 .. note::
    If a setting is set through an environment variable and any other changes are made in the System Console, the value stored of the environment variable will be written back to the ``config.json`` as that setting's value.
 
 .. warning::
    Database connection strings for the database read and search replicas need to be formatted using `URL encoding <https://www.w3schools.com/tags/ref_urlencode.asp>`__. Incorrectly formatted strings may cause some characters to terminate the string early, resulting in issues when the connection string is parsed.
+   
+**Load Custom Configuration Defaults**
+
+Starting from Mattermost version 5.30, you can load a set of custom configuration defaults using an environment variable.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:39]:
```suggestion
Starting from Mattermost v5.30, you can load a set of custom configuration defaults using an environment variable.
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
One, non-blocking, suggestion. Otherwise LGTM!

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -26,13 +26,21 @@ The name of the environment variable for any setting can be derived from the nam
...
 
 .. note::
    If a setting is set through an environment variable and any other changes are made in the System Console, the value stored of the environment variable will be written back to the ``config.json`` as that setting's value.
 
 .. warning::
    Database connection strings for the database read and search replicas need to be formatted using `URL encoding <https://www.w3schools.com/tags/ref_urlencode.asp>`__. Incorrectly formatted strings may cause some characters to terminate the string early, resulting in issues when the connection string is parsed.
+   
+**Load Custom Configuration Defaults**
+
+Starting from Mattermost v5.30, you can load a set of custom configuration defaults using an environment variable.
```
Comment (mgdelacroix) [inline source/administration/config-settings.rst:39]: Adding a bit more information about when this configuration applies

```suggestion
Starting from Mattermost v5.30, you can load a set of custom configuration defaults using an environment variable. This custom configuration will apply only if the values are not already present in the current server configuration.
```

File: source/administration/config-settings.rst [hunk 3]
```diff
@@ -26,13 +26,21 @@ The name of the environment variable for any setting can be derived from the nam
...
+
+1. Create a ``custom.json`` file that contains the custom configuration defaults.
```
Comment (mgdelacroix) [inline source/administration/config-settings.rst:41]: nit: making explicit that the file name is an example

```suggestion
1. Create a JSON file that contains the custom configuration defaults, for example ``custom.json``.
```

File: source/administration/config-settings.rst [hunk 4]
```diff
@@ -26,13 +26,21 @@ The name of the environment variable for any setting can be derived from the nam
...
+2. Edit the ``config.json`` file to include the new environment variable ``MM_CUSTOM_DEFAULTS_PATH=custom.json``.
```
Comment (mgdelacroix) [inline source/administration/config-settings.rst:42]: Reference the server startup process:

```suggestion
2. When starting the server, point the custom defaults environment variable to the defaults file: ``MM_CUSTOM_DEFAULTS_PATH=custom.json``.
```

**Comment by cwarnermm** [comment]
Thank you so much @mgdelacroix for your excellent documentation suggestions and improvements.

**Review by mgdelacroix — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]
Looks good to me.

**Pull request merged**